examples/zkevm-demo/src/main.rs

```diff
@@ -82,22 +83,17 @@ async fn main() {
     let zkdb = trace_db.create_zkdb();
 
     info!("Running zkvm...");
-    let env = ExecutorEnv::builder()
+    let exec_env = ExecutorEnv::builder()
         .add_input(&to_vec(&env).unwrap())
         .add_input(&to_vec(&zkdb).unwrap())
         .build()
         .unwrap();
-    let mut exec = Executor::from_elf(env, EVM_ELF).unwrap();
-    let segment_dir = tempdir().unwrap();
-    let session = exec
-        .run_with_callback(|segment| {
-            Ok(Box::new(FileSegmentRef::new(
-                &segment,
-                &segment_dir.path(),
-            )?))
```

Comment:
If we want this again in the future, maybe we could use executor env as a way to configure `run_with_callback` to use files. We certainly don't need to implement this right now though!

What's nice about this is that it's saving memory for large computations and it would be nice to give the user the ability to opt-into this during development if they run a large proof during development.

Reply:
I've been thinking a similar thing. It'd be nice to just specify the kind of standard handling in the ExecutorEnv. I think we should switch the default to use temporary files but we should allow the user to configure where the directory lives.